codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -96,6 +104,71 @@ public DefaultHttp2Connection(boolean server) {
         streamMap.put(connectionStream.id(), connectionStream);
     }
 
+    /**
+     * Determine if {@link #close(Promise)} has been called and no more streams are allowed to be created.
+     */
+    final boolean isClosed() {
+        return closePromise != null;
+    }
+
+    @Override
+    public Future<Void> close(final Promise<Void> promise) {
```

Comment:
It makes unit testing easier and there is no reason why the code has to have a `ChannelPromise` if it can work with a `Promise<Void>`.